astropy/config/paths.py

```diff
@@ -19,6 +20,46 @@
 ]
 
 
+class _DirectoryType(Enum):
+    CONFIG = auto()
+    CACHE = auto()
+
+
+def _get_dir_path(rootname: str, dirtype: _DirectoryType) -> Path:
+    match dirtype:
+        case _DirectoryType.CONFIG:
+            cls = set_temp_config
+            fallback = "config"
+        case _DirectoryType.CACHE:
+            cls = set_temp_cache
+            fallback = "cache"
+        case _:
+            raise ValueError(f"Unexpected {dirtype=}")
```

Comment:
Typo or another fancy type checking syntax I don't understand? 🧐 

```suggestion
            raise ValueError(f"Unexpected dirtype={dirtype}")
```

Reply:
neither !
This is a "debug fstring" from Python 3.8
see https://docs.python.org/3/whatsnew/3.8.html#f-strings-support-for-self-documenting-expressions-and-debugging

Reply:
Oh TIL... 😆  Thanks for the clarification!